main.js

```diff
@@ -49,31 +49,43 @@ function validateUserInput() {
 
 function displayCards() {
   ideaCardGrid.innerHTML = "";
+  var shouldShowAllIdeas = showStarredIdeasButton.innerText === `Show Starred Ideas`;
+  var shouldShowStarredIdeas = showStarredIdeasButton.innerText === `Show All Ideas`;
   for (var i = 0; i < ideas.length; i++) {
-    var starSource;
-    if (ideas[i].star) {
-      starSource = "./assets/star-active.svg";
-    } else {
-      starSource = "./assets/star.svg";
+    if (shouldShowAllIdeas) {
+      ideaCardGrid.innerHTML += generateInnerHTML(ideas[i]);
+    } else if (shouldShowStarredIdeas && ideas[i].star) {
+      ideaCardGrid.innerHTML += generateInnerHTML(ideas[i]);
     }
```

Comment:
I created the variables shouldShowAllIdeas/ shouldShowStarredIdeas and assigned them to the values of the showStarredIdeasButton.innerText for better readability. This for loop will iterate through our ideas array element by element and execute based on the conditionals on line 55 and 57. On lines 55-56,  shouldShowAllIdeas will generate innerHTML for our ideaCard grid for every idea in our ideas array. On line 57, if we want shouldShowStarredIdeas then by definition we only want elements in our ideas array whose star property is true. If both of these are satisfied, then this will generate innerHTML for our ideaCard grid for every idea in our ideas array whose star property is true.